invenio_rdm_records/records/dumpers.py

```diff
@@ -57,5 +65,59 @@ def load(self, data, record_cls):
         parent_data = dict_lookup(data, self.keys, parent=True)
 
         # `None` covers the cases where exceptions were raised in _dump
-        parent_data.pop(f"{self.key}_start", None)
-        parent_data.pop(f"{self.key}_end", None)
+        parent_data.pop(self.range_key, None)
+
+
+class EDTFListDumperExt(ElasticsearchDumperExt):
+    """Elasticsearch dumper extension for support of EDTF date lists.
+
+    It iterates over the items at the specified field in the
+    document and adds dictionaries holding the values
+    :code:`{"lte": lower_strict, "gte": upper_strict}`, as parsed
+    from their EDTF fields (specified by the key).
+    These values correspond to the lower strict and upper strict bounds.
+    They are required for sorting.
+    In an ascending sort (most recent last), sort by publication_date_start
+    to get a natural sort order. In a descending sort (most recent first),
+    sort by publication_date_end.
+    """
+
+    def __init__(self, list_field, key):
+        """Constructor.
+
+        :param list_field: dot separated path to the array to process.
+        :param key: name of the EDTF field in each array item.
+        """
+        super(EDTFListDumperExt, self).__init__()
+        self.keys = parse_lookup_key(list_field)
+        self.key = key
+        self.range_key = "{}_range".format(self.key)
+
+    def dump(self, record, data):
+        """Dump the data."""
+        try:
+            date_list = dict_lookup(data, self.keys, parent=False)
```

Comment:
question: why do we have to use `parent=False` here and in the single date dumper we can use `parent=True`? It has to do with how the `self.key` is assigned I guess. I am just wondering if this can be refactored into a parent class since both the constructor and the code inside the loop is duplicated in both classes.

Reply:
In ``publication_date``, we need access to add the extra keys at the same level as publication date:

```
{
  "publication_date": "..."
  "publication_date_range": ...
}
```
For dates, we're getting the list, and need to add the objects inside each item, so there's no need to get the parent:

```
[
{
  "date": "..."
  "date_range": ...
}
]
```